File: app/src/main/java/org/scp/gymlog/model/Muscle.kt
package org.scp.gymlog.model

import androidx.annotation.ColorRes
import androidx.annotation.DrawableRes
import androidx.annotation.StringRes
import org.scp.gymlog.R
import org.scp.gymlog.room.EntityMappable
import org.scp.gymlog.room.entities.MuscleEntity

enum class Muscle(
	@StringRes val text: Int,
	@StringRes val textShort: Int,
	@DrawableRes val icon: Int,
	@ColorRes val color: Int
) : EntityMappable<MuscleEntity> {

	PECTORAL (
		text = R.string.group_pectoral,
		textShort = R.string.group_short_pectoral,
		icon = R.drawable.muscle_pectoral,
		color = R.color.pectoral
	),
	UPPER_BACK (
		text = R.string.group_upper_back,
		textShort = R.string.group_short_upper_back,
		icon = R.drawable.muscle_upper_back,
		color = R.color.upper_back
	),
	LOWER_BACK (
		text = R.string.group_lower_back,
		textShort = R.string.group_short_lower_back,
		icon = R.drawable.muscle_lower_back,
		color = R.color.lower_back
	),
	DELTOID (
		text = R.string.group_deltoid,
		textShort = R.string.group_short_deltoid,
		icon = R.drawable.muscle_deltoid,
		color = R.color.deltoid
	),
	TRAPEZIUS (
		text = R.string.group_trapezius,
		textShort = R.string.group_short_trapezius,
		icon = R.drawable.muscle_trapezius,
		color = R.color.trapezius
	),
	BICEPS (
		text = R.string.group_biceps,
		textShort = R.string.group_short_biceps,
		icon = R.drawable.muscle_biceps,
		color = R.color.biceps
	),
	TRICEPS (
		text = R.string.group_triceps,
		textShort = R.string.group_short_triceps,
		icon = R.drawable.muscle_triceps,
		color = R.color.triceps
	),
	FOREARM (
		text = R.string.group_forearm,
		textShort = R.string.group_short_forearm,
		icon = R.drawable.muscle_forearm,
		color = R.color.forearm
	),
	QUADRICEPS (
		text = R.string.group_quadriceps,
		textShort = R.string.group_short_quadriceps,
		icon = R.drawable.muscle_quadriceps,
		color = R.color.quadriceps
	),
	HAMSTRINGS (
		text = R.string.group_hamstrings,
		textShort = R.string.group_short_hamstrings,
		icon = R.drawable.muscle_hamstring,
		color = R.color.hamstrings
	),
	GLUTES (
		text = R.string.group_glutes,
		textShort = R.string.group_short_glutes,
		icon = R.drawable.muscle_glutes,
		color = R.color.glutes
	),
	CALVES (
		text = R.string.group_calves,
		textShort = R.string.group_short_calves,
		icon = R.drawable.muscle_calves,
		color = R.color.calves
	),
	ABDOMINALS (
		text = R.string.group_abdominals,
		textShort = R.string.group_short_abdominals,
		icon = R.drawable.muscle_abdominals,
		color = R.color.abdominals
	),
	CARDIO (
		text = R.string.group_cardio,
		textShort = R.string.group_short_cardio,
		icon = R.drawable.muscle_cardio,
		color = R.color.cardio
	);

	val id = ordinal + 1

	override fun toEntity(): MuscleEntity {
		val entity = MuscleEntity()
		entity.muscleId = id
		return entity
	}
}

File: private/js/model/muscle.ts
export interface Muscle {
  id: number,
  name: string,
  color: string,
}

export const MUSCLES: Muscle[] = [
  { id: 1,  name: 'Chest (Pectoral)',    color: '#9675CE' },
  { id: 2,  name: 'Upper Back',          color: '#1073AE' },
  { id: 3,  name: 'Lower Back',          color: '#4DD0E2' },
  { id: 4,  name: 'Shoulders (Deltoid)', color: '#E05E55' },
  { id: 5,  name: 'Trapezius',           color: '#F05F92' },
  { id: 6,  name: 'Biceps',              color: '#8DCA1A' },
  { id: 7,  name: 'Triceps',             color: '#01BFA5' },
  { id: 8,  name: 'Forearm',             color: '#71BA6E' },
  { id: 9,  name: 'Quadriceps',          color: '#FFD33F' },
  { id: 10, name: 'Hamstrings',          color: '#F9A825' },
  { id: 11, name: 'Glutes',              color: '#F96825' },
  { id: 12, name: 'Calves',              color: '#B7AB6E' },
  { id: 13, name: 'Abdominals',          color: '#CA62F0' },
  { id: 14, name: 'Cardio',              color: '#AAABAA' },
];